Tokens & NFTs contracts and examples

ogabrielides opened the pull request.

abresas commented on examples/tokens/Tokens.sol:
```diff
@@ -0,0 +1,40 @@
+    string  public symbol;
+    uint256 public totalSupply;
+
+    mapping(address => uint256) private balances;
```
tokens must support negative balances as in fast pay

abresas commented on examples/nfts/NFTs.sol:
```diff
@@ -0,0 +1,85 @@
+    event Transferred(uint256 indexed tokenId, address indexed from, address indexed to);
+
+    function mint(uint256 tokenId, string calldata uri) external {
+        requireQuorum(tokens[tokenId].owner == address(0), "already minted");
```
It's better to have all NFTs initially belong to a particular address. So, if the token address is 0, then only the `tx.origin` can "mint", transferring it to another address.

Otherwise, there could be a race causing locks.

abresas commented on examples/nfts/NFTs.sol:
```diff
@@ -0,0 +1,85 @@
+        emit Transferred(id, from, to);
+    }
+
+    function _checkOnERC721Received(
```
all this seems very complicated, is there a reason it is like this?

ogabrielides commented:
@abresas Addressed issues.
How about now?

A bit concerned on this getter. Does this return error on a validator that hasn't seen yet the minted tokenId?
``function tokenURI(uint256 tokenId) external view returns (string memory) {
        return tokensUri[tokenId];
    }``
